src/language-html/parser-html.js

```diff
@@ -132,21 +133,23 @@ function ngHtmlParser(
   }
 
   const addType = (node) => {
-    if (node instanceof Attribute) {
-      node.type = "attribute";
-    } else if (node instanceof CDATA) {
-      node.type = "cdata";
-    } else if (node instanceof Comment) {
-      node.type = "comment";
-    } else if (node instanceof DocType) {
-      node.type = "docType";
-    } else if (node instanceof Element) {
-      node.type = "element";
-    } else if (node instanceof Text) {
-      node.type = "text";
-    } else {
-      throw new Error(`Unexpected node ${JSON.stringify(node)}`);
-    }
+    // using a ternary to avoid JSDoc type issues
+    node.type =
+      node instanceof Attribute
+        ? "attribute"
+        : node instanceof CDATA
+        ? "cdata"
+        : node instanceof Comment
+        ? "comment"
+        : node instanceof DocType
+        ? "docType"
+        : node instanceof Element
+        ? "element"
+        : node instanceof Text
+        ? "text"
+        : (() => {
+            throw new Error(`Unexpected node ${JSON.stringify(node)}`);
+          })();
```

Comment:
And this

Reply:
Rationale is in the comment. Any better ideas?

Reply:
i tried to add to this function, seems working

```js
  /**
  * @typedef {import('angular-html-parser/lib/compiler/src/ml_parser/ast').Node} Node
  * @typedef {object} WithType
  * @property {string} type
  * @typedef {Node & WithType} NodeWithType
  * @param {NodeWithType} node
  */
```

, but I also tried

```js
  /**
  * @typedef {object} WithType
  * @property {string} type
  * @param {WithType} node
  */
```

also work, I don't really know what those things mean...

Reply:
I think `@typedef {object}` is very similar to `@typedef {any}`, they may even be the same thing. (I recall @thorn0 said something about this in another review.) I would actually favor using `any`, to make it easier for us to find and improve at some point in the future.

I tried this to be a little more specific but it did not work:

```js
/**
 * @typedef {import('angular-html-parser/lib/compiler/src/ml_parser/ast').Node} WithType
 * @property {string} type
 * @param {WithType} node
 */
```

I am now thinking it would be best to keep the JSDoc typedefs near the beginning of the module, will work on this.

Reply:
But I think it's just Node with an extra type, I just don't know how to do it.

Reply:
Or maybe it's Element or something, there are several types , one or some together should work